vignettes/concepts_conventions.Rmd

```diff
@@ -0,0 +1,285 @@
+converted to R `NA`s using `{admiral}`'s `convert_blanks_to_na()` function, e.g.
+
+```r
+dm <- haven::read_sas("dm.sas7bdat") %>% 
+  convert_blanks_to_na()
+```
+
+Note that any logical operator being applied to an `NA` value *always* returns `NA` rather than `TRUE` or `FALSE`.
+
+```{r}
+visits <- c("Baseline", NA, "Screening", "Week 1 Day 7")
+visits != "Baseline"
+```
+
+The only exception is `is.na()` which returns `TRUE` if the input is `NA`.
+
+```{r}
+is.na(visits)
+```
+
+Thus, to filter all visits which are not `"Baseline"` the following condition would need to be used.
+
+```{r}
+visits != "Baseline" | is.na(visits)
+```
+
+Also note that most aggregation functions, like `mean()` or `max()`, also return `NA` if any element of the input vector is missing.
+
+```{r}
+mean(c(1, NA, 2))
+```
+
+To avoid this behavior one has to explicitly set `na.rm = TRUE`.
+
+```{r}
+mean(c(1, NA, 2), na.rm = TRUE)
+```
+
+This is very important to keep in mind when using `{admiral}`'s aggregation functions such as `derive_summary_records()`.
+
+For handling of `NA`s in sorting variables see [Sort Order](generic.html#sort_order).
+
+# Expressions in Scripts {#exprs}
+
+## Quoting and Unquoting: Introducing `expr()`, `exprs()`, `!!` and `!!!`
+
+### `expr()` and `exprs()`
+
+`expr()` is a function from the `{rlang}` package, which is used to create an **expression**. The expression is not evaluated 
+- rather, it is passed on to the derivation function which evaluates it in its own environment. `exprs()` is the plural version of 
+`expr()`, so it accepts multiple comma-separated items and returns a list of expressions. 
+
+```{r, eval = TRUE}
+library(rlang)
+
+adae <- data.frame(USUBJID = "XXX-1", AEDECOD = "HEADACHE")
+
+# Return the adae object
+adae
+
+# Return an expression
+expr(adae)
+```
+
+When used within the contest of an `{admiral}` derivation function, `expr()` and `exprs()`  allow the function to evaluate the 
+expressions in the context of the input dataset. As an example, `expr()` and `exprs()` allow users to pass variable names of datasets 
+to the function without wrapping them in quotation marks. 
+
+The expressions framework is powerful because users are able to intuitively "inject code" into `admiral` functions
+(through the function parameters) using very similar syntax as if they were writing open code, with the exception possibly being an
+outer `exprs()` wrapper.  For instance, in the `derive_vars_merged()` call below, the user is merging `adsl` with `ex` and is able
+to filter `ex` prior to the merge using an expression passed to the `filter_add` parameter. Because `filter_add` accepts expressions, 
+the user has full power to filter their dataset as they please. In the same vein, the user is able to create any new variables they 
+wish after the merge using the `new_vars` argument, to which they pass a list of expressions containing "standard" R code.
```

Comment:
```suggestion
The expressions framework is powerful because users are able to intuitively "inject code" into `admiral` functions
(through the function parameters) using very similar syntax as if they were writing open code, with the exception possibly being an
outer `exprs()` wrapper.  For instance, in the `derive_vars_merged()` call below, the user is merging `ADSL` with `EX` and is able
to filter `EX` prior to the merge using an expression passed to the `filter_add` parameter. Because `filter_add` accepts expressions, 
the user has full power to filter their dataset as they please. In the same vein, the user is able to create any new variables they 
wish after the merge using the `new_vars` argument, to which they pass a list of expressions containing "standard" R code.
```

Reply:
again this portion is referencing the actual `adsl` and `ex` objects in the code rather than the general `ADSL` and `EX` datasets